syntax/elixir.vim

```diff
@@ -160,7 +160,7 @@ syn match  elixirExceptionDeclaration   "[^[:space:];#<]\+"        contained con
 syn match  elixirCallbackDeclaration    "[^[:space:];#<,()\[\]]\+" contained contains=elixirFunctionDeclaration             skipwhite skipnl
 
 " ExUnit
-syn match  elixirExUnitMacro "\(^\s*\)\@<=\<\(test\|describe\|setup\|setup_all\|on_exit\|doctest\)\>"
+syn match  elixirExUnitMacro "\(^\s*\)\@<=\<\(test\|describe\|setup\|setup_all\|on_exit\|doctest\)\> "
```

Comment:
Another idea would be to use negative lookaround for the `:` character to make sure it's not an atom. I've done that frequently in the indent code (example: https://github.com/elixir-editors/vim-elixir/blob/master/autoload/elixir/indent.vim#L112). That would avoid the parenthesis change

Reply:
But that would leave us with `setup()` beinging highlighted as macro.

Basically I try to distinguish between 3 cases: `test ...`, `test()`, and `test:`. I know the issue is just about the `test:` case, but I guess functions being called `test()` are not too uncommon. 
So.. assuming that the ExUnit macro is always followed by space, we can either add a space at the end of the regex or use that negative lookaround like `\[:\(]\@!`. The outcome is the same.

I guess the question if we think that functions named `test()` are more common than the ExUnit macro `test` with parentheses. (I never saw the latter so far.)

Reply:
Hmm, I somehow missed that response. I agree! Closing this in favor of #432